### ISSUE TITLE
False positive error "incompatible type"

### ISSUE THREAD

**Issue by Mirraz** [python/mypy#13566]
For this code:
```python
from typing import Any, Generic, TypeVar

T = TypeVar('T')


class EtcdValueWatcherTyped(Generic[T]):
    def __init__(self):
        pass


class EtcdValueWatcher(EtcdValueWatcherTyped[dict[str, Any]]):
    pass


class SettingsWatcher(EtcdValueWatcher):
    def __init__(self, arg1: str):
        self.__arg1 = arg1
        EtcdValueWatcher.__init__(self)


def main():
    watcher = SettingsWatcher('/v1/qwe')
```
mypy outputs error:
```
test.py: note: In member "__init__" of class "SettingsWatcher":
test.py:19: error: Argument 1 to "__init__" has incompatible type "SettingsWatcher"; expected "EtcdValueWatcherTyped[T]"  [arg-type]
            EtcdValueWatcher.__init__(self)
                                      ^
Found 1 error in 1 file (checked 1 source file)
```
which is false positive.

- Mypy version used: `mypy 0.971 (compiled: yes)`
- Mypy command-line flags: `mypy --config-file ./mypy.ini ./test.py`
- Mypy configuration options from `mypy.ini` (and other config files):
```
[mypy]
ignore_missing_imports = True
show_error_codes = True
show_error_context = True
pretty = True
no_site_packages = False
```
- Python version used: Python 3.10.6
- Operating system and version: Fedora 35

**Comment by asqui** [comment]
We've run in to this problem too...

Is there an easy way to work around this in the meantime until a fix is released?

**Comment by leandro-lucarella-frequenz** [comment]
Same here, I think this is a more reduced test case:

```py
from typing import Generic, TypeVar

T = TypeVar("T", float, int)


class C(Generic[T]):
    def __init__(self, i: T) -> None:
        self.i: T = i


class B(C[T]):
    def __init__(self, i: T) -> None:
        super().__init__(i)
```

[mypy-playground](https://mypy-play.net/?mypy=latest&python=3.11&flags=show-error-context%2Cshow-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cno-implicit-optional%2Cno-implicit-reexport%2Cno-strict-optional%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=0d988d1bb67e609cabd6521da88056ed)

**Comment by pmantion** [comment]
I ran into this issue as well.
